File: yavel/src/androidTest/java/com/awolity/yavel/YavelTest.java
package com.awolity.yavel;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import androidx.test.InstrumentationRegistry;
import androidx.test.runner.AndroidJUnit4;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

@SuppressWarnings("unused")
@RunWith(AndroidJUnit4.class)
public class YavelTest {

    private Yavel yavel;
    private final String KEY_ALIAS = "yavel key alias";

    @Before
    public void setup() throws Exception {
        yavel = Yavel.Companion.get(KEY_ALIAS);
    }

    @Test
    public void stringTest() throws Exception {
        String str = "whatever";
        String encryptedString = yavel.encryptString(str);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertEquals(str, yavel.decryptString(encryptedString));
    }

    @Test
    public void intTest() throws Exception {
        int value = 12345;
        String encryptedInt = yavel.encryptInt(value);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertEquals(value, yavel.decryptInt(encryptedInt));
    }

    @Test
    public void booleanTest() throws Exception {
        String encryptedValue = yavel.encryptBoolean(true);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertTrue(yavel.decryptBoolean(encryptedValue));
    }

    @Test
    public void booleanTest2() throws Exception {
        String encryptedValue = yavel.encryptBoolean(false);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        Assert.assertFalse(yavel.decryptBoolean(encryptedValue));
    }

    @Test
    public void longTest() throws Exception {
        long value = 12345678L;
        String encryptedValue = yavel.encryptLong(value);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertEquals(value, yavel.decryptLong(encryptedValue));
    }

    @Test
    public void floatTest() throws Exception {
        float value = 1234.5678F;
        String encryptedValue = yavel.encryptFloat(value);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertEquals(value, yavel.decryptFloat(encryptedValue), 0.00000001);
    }

    @Test
    public void doubleTest() throws Exception {
        double value = 1234.5678;
        String encryptedValue = yavel.encryptDouble(value);
        yavel = null;
        yavel = Yavel.Companion.get(KEY_ALIAS);
        assertEquals(value, yavel.decryptDouble(encryptedValue), 0.00000001);
    }
}

File: yavel/src/androidTest/java/com/awolity/yavel/YavelTest.kt
package com.awolity.yavel

import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

import androidx.test.runner.AndroidJUnit4

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue

@RunWith(AndroidJUnit4::class)
class YavelTest {

    private var yavel: Yavel? = null
    private val KEY_ALIAS = "yavel key alias"

    @Before
    @Throws(Exception::class)
    fun setup() {
        yavel = Yavel[KEY_ALIAS]
    }

    @Test
    @Throws(Exception::class)
    fun stringTest() {
        val str = "whatever"
        val encryptedString = yavel!!.encryptString(str)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertEquals(str, yavel!!.decryptString(encryptedString))
    }

    @Test
    @Throws(Exception::class)
    fun intTest() {
        val value = 12345
        val encryptedInt = yavel!!.encryptInt(value)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertEquals(value.toLong(), yavel!!.decryptInt(encryptedInt).toLong())
    }

    @Test
    @Throws(Exception::class)
    fun booleanTest() {
        val encryptedValue = yavel!!.encryptBoolean(true)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertTrue(yavel!!.decryptBoolean(encryptedValue))
    }

    @Test
    @Throws(Exception::class)
    fun booleanTest2() {
        val encryptedValue = yavel!!.encryptBoolean(false)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        Assert.assertFalse(yavel!!.decryptBoolean(encryptedValue))
    }

    @Test
    @Throws(Exception::class)
    fun longTest() {
        val value = 12345678L
        val encryptedValue = yavel!!.encryptLong(value)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertEquals(value, yavel!!.decryptLong(encryptedValue))
    }

    @Test
    @Throws(Exception::class)
    fun floatTest() {
        val value = 1234.5678f
        val encryptedValue = yavel!!.encryptFloat(value)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertEquals(value.toDouble(), yavel!!.decryptFloat(encryptedValue).toDouble(), 0.00000001)
    }

    @Test
    @Throws(Exception::class)
    fun doubleTest() {
        val value = 1234.5678
        val encryptedValue = yavel!!.encryptDouble(value)
        yavel = null
        yavel = Yavel[KEY_ALIAS]
        assertEquals(value, yavel!!.decryptDouble(encryptedValue), 0.00000001)
    }
}
